internal/provider/resource_tfe_project.go

```diff
@@ -31,6 +32,11 @@ func resourceTFEProject() *schema.Resource {
 			"name": {
 				Type:     schema.TypeString,
 				Required: true,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(3, 36),
+					validation.StringMatch(regexp.MustCompile(`\A[\w\_\- ]+\z`),
```

Comment:
This is that regexp adjustment — excluding space from the first and last character classes. (Also, underscores are included in \w, so they don't need to be explicitly added to the class.)

```suggestion
					validation.StringMatch(regexp.MustCompile(`\A[\w\-][\w\- ]+[\w\-]\z`),
```